diff --git a/src/app/app-routing.module.ts b/src/app/app-routing.module.ts
index bee8416f..dad32467 100644
--- a/src/app/app-routing.module.ts
+++ b/src/app/app-routing.module.ts
@@ -33,6 +33,7 @@ import { AccountComponent as OrgAccountComponent } from './organizations/setting
 import { OrganizationBillingComponent } from './organizations/settings/organization-billing.component';
 import { OrganizationSubscriptionComponent } from './organizations/settings/organization-subscription.component';
 import { SettingsComponent as OrgSettingsComponent } from './organizations/settings/settings.component';
+import { SsoComponent as OrgSsoComponent } from './organizations/settings/sso.component';
 import {
     TwoFactorSetupComponent as OrgTwoFactorSetupComponent,
 } from './organizations/settings/two-factor-setup.component';
@@ -412,6 +413,7 @@ const routes: Routes = [
                 children: [
                     { path: '', pathMatch: 'full', redirectTo: 'account' },
                     { path: 'account', component: OrgAccountComponent, data: { titleId: 'myOrganization' } },
+                    { path: 'sso', component: OrgSsoComponent, data: { titleId: 'sso' } },
                     { path: 'two-factor', component: OrgTwoFactorSetupComponent, data: { titleId: 'twoStepLogin' } },
                     {
                         path: 'billing',
diff --git a/src/app/app.module.ts b/src/app/app.module.ts
index bafc22a0..938db7ee 100644
--- a/src/app/app.module.ts
+++ b/src/app/app.module.ts
@@ -67,6 +67,7 @@ import { DownloadLicenseComponent } from './organizations/settings/download-lice
 import { OrganizationBillingComponent } from './organizations/settings/organization-billing.component';
 import { OrganizationSubscriptionComponent } from './organizations/settings/organization-subscription.component';
 import { SettingsComponent as OrgSettingComponent } from './organizations/settings/settings.component';
+import { SsoComponent as OrgSsoComponent } from './organizations/settings/sso.component';
 import {
     TwoFactorSetupComponent as OrgTwoFactorSetupComponent,
 } from './organizations/settings/two-factor-setup.component';
@@ -347,6 +348,7 @@ registerLocaleData(localeZhTw, 'zh-TW');
         NavbarComponent,
         OptionsComponent,
         OrgAccountComponent,
+        OrgSsoComponent,
         OrgAddEditComponent,
         OrganizationBillingComponent,
         OrganizationPlansComponent,
diff --git a/src/app/organizations/settings/sso.component.html b/src/app/organizations/settings/sso.component.html
index 41d0e89e..c1f2ccf5 100644
--- a/src/app/organizations/settings/sso.component.html
+++ b/src/app/organizations/settings/sso.component.html
@@ -5,38 +5,38 @@
     <i class="fa fa-spinner fa-spin text-muted" title="{{'loading' | i18n}}" aria-hidden="true"></i>
     <span class="sr-only">{{'loading' | i18n}}</span>
 </div>
-<form *ngIf="org && !loading" #form (ngSubmit)="submit()" [appApiAction]="formPromise" ngNativeValidate>
+<form *ngIf="ssoConfig && !loading" #form (ngSubmit)="submit()" [appApiAction]="formPromise" ngNativeValidate>
     <div class="row">
         <div class="col-12">
             <div class="form-group">
                 <label for="enabled">{{'enabled' | i18n}}</label>
-                <input id="enabled" class="form-control" type="checkbox" name="Enabled" [(ngModel)]="org.useSso" [disabled]="selfHosted">
+                <input id="enabled" class="form-control" type="checkbox" name="Enabled" [(ngModel)]="ssoConfig.useSso" [disabled]="selfHosted">
             </div>
             <h2>OpenId Connect Configuration</h2>
             <div class="form-group">
                 <label for="callbackPath">{{'callbackPath' | i18n}}</label>
-                <input id="callbackPath" class="form-control" type="text" name="Callback Path" [(ngModel)]="org.callbackPath"
+                <input id="callbackPath" class="form-control" type="text" name="Callback Path" [(ngModel)]="ssoConfig.callbackPath"
                     [disabled]="selfHosted">
             </div>
             <div class="form-group">
                 <label for="signedOutCallbackPath">{{'signedOutCallbackPath' | i18n}}</label>
                 <input id="signedOutCallbackPath" class="form-control" type="text" name="Signed Out Callback Path"
-                    [(ngModel)]="org.signedOutCallbackPath" [disabled]="selfHosted">
+                    [(ngModel)]="ssoConfig.signedOutCallbackPath" [disabled]="selfHosted">
             </div>
             <div class="form-group">
                 <label for="authority">{{'authority' | i18n}}</label>
                 <input id="authority" class="form-control" type="text" name="Authority"
-                    [(ngModel)]="org.authority" [disabled]="selfHosted">
+                    [(ngModel)]="ssoConfig.authority" [disabled]="selfHosted">
             </div>
             <div class="form-group">
                 <label for="clientId">{{'clientId' | i18n}}</label>
                 <input id="authority" class="form-control" type="text" name="Client ID"
-                    [(ngModel)]="org.clientId" [disabled]="selfHosted">
+                    [(ngModel)]="ssoConfig.clientId" [disabled]="selfHosted">
             </div>
             <div class="form-group">
                 <label for="clientSecret">{{'clientSecret' | i18n}}</label>
                 <input id="clientSecret" class="form-control" type="password" name="Client Secret"
-                    [(ngModel)]="org.clientSecret" [disabled]="selfHosted">
+                    [(ngModel)]="ssoConfig.clientSecret" [disabled]="selfHosted">
             </div>
         </div>
     </div>
@@ -45,7 +45,7 @@
         <span>{{'save' | i18n}}</span>
     </button>
 </form>
-<div *ngIf="!org || loading">
+<div *ngIf="!ssoConfig || loading">
     <i class="fa fa-spinner fa-spin text-muted" title="{{'loading' | i18n}}" aria-hidden="true"></i>
     <span class="sr-only">{{'loading' | i18n}}</span>
 </div>
diff --git a/src/app/organizations/settings/sso.component.ts b/src/app/organizations/settings/sso.component.ts
index f40a54f2..5eeef132 100644
--- a/src/app/organizations/settings/sso.component.ts
+++ b/src/app/organizations/settings/sso.component.ts
@@ -17,7 +17,7 @@ import { SyncService } from 'jslib-common/abstractions/sync.service';
 
 import { OrganizationSsoUpdateRequest } from 'jslib-common/models/request/organizationSsoUpdateRequest';
 
-import { OrganizationResponse } from 'jslib-common/models/response/organizationResponse';
+import { SsoConfigResponse } from 'jslib-common/models/response/ssoConfigResponse';
 
 import { ModalComponent } from '../../modal.component';
 
@@ -28,7 +28,7 @@ import { ModalComponent } from '../../modal.component';
 export class SsoComponent {
     selfHosted = false;
     loading = true;
-    org: OrganizationResponse;
+    ssoConfig: SsoConfigResponse;
     formPromise: Promise<any>;
 
     private organizationId: string;
@@ -45,7 +45,7 @@ export class SsoComponent {
         this.route.parent.parent.params.subscribe(async params => {
             this.organizationId = params.organizationId;
             try {
-                this.org = await this.apiService.getOrganization(this.organizationId);
+                this.ssoConfig = await this.apiService.getSsoConfig(this.organizationId);
             } catch { }
         });
         this.loading = false;
@@ -54,12 +54,12 @@ export class SsoComponent {
     async submit() {
         try {
             const request = new OrganizationSsoUpdateRequest();
-            request.useSso = this.org.useSso;
-            request.callbackPath = this.org.callbackPath;
-            request.signedOutCallbackPath = this.org.signedOutCallbackPath;
-            request.authority = this.org.authority;
-            request.clientId = this.org.clientId;
-            request.clientSecret = this.org.clientSecret;
+            request.useSso = this.ssoConfig.useSso;
+            request.callbackPath = this.ssoConfig.callbackPath;
+            request.signedOutCallbackPath = this.ssoConfig.signedOutCallbackPath;
+            request.authority = this.ssoConfig.authority;
+            request.clientId = this.ssoConfig.clientId;
+            request.clientSecret = this.ssoConfig.clientSecret;
 
             this.formPromise = this.apiService.putOrganizationSso(this.organizationId, request).then(() => {
                 return this.syncService.fullSync(true);
